core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -300,6 +330,47 @@ private DetachedPlugin(String shortName, String splitWhen, String requireVersion
             this.requireVersion = requireVersion;
         }
 
+        /**
+         * Get the short name of the plugin.
+         * @return The short name of the plugin.
+         * @since FIXME
+         */
+        public String getShortName() {
+            return shortName;
+        }
+
+        /**
+         * Get the Jenkins version from which the plugin was detached.
+         * @return The Jenkins version from which the plugin was detached.
+         * @since FIXME
+         */
+        public VersionNumber getSplitWhen() {
+            return splitWhen;
+        }
+
+        /**
+         * Get the minimum required version of the plugin.
+         * @return The minimum required version of the plugin.
+         * @since FIXME
+         */
+        public String getRequireVersion() {
+            return requireVersion;
+        }
+
+        /**
+         * Convert a collection of DetachedPlugins to a Set of plugins names.
+         * @param detachedPlugins The collection of DetachedPlugins.
+         * @return The list of plugin names.
+         * @since FIXME
+         */
+        public static Set<String> toPluginNameList(@Nonnull Collection<ClassicPluginStrategy.DetachedPlugin> detachedPlugins) {
+            Set<String> pluginNames = new LinkedHashSet<>();
+            for (ClassicPluginStrategy.DetachedPlugin detachedPlugin : detachedPlugins) {
+                pluginNames.add(detachedPlugin.getShortName());
+            }
+            return pluginNames;
```

Comment:
Well, "trivial" for one person can be obscure to others. I'm not clever enough to be able to look at your alternative and immediately twig "oh yes ... that's just extracting a list of plugin names", so we'll have to agree to disagree on that one :smile:

Anyway ... your last point is what's most relevant to me because yes, that (and the method before, which I also added) do not seem to be used anymore so we can remove them. I must have been using them at an earlier stage in this PR.